swap/src/main.rs

```diff
@@ -83,6 +77,11 @@ async fn main() -> Result<()> {
 
             let swap_id = Uuid::new_v4();
 
+            info!(
+                "Swap sending {} and receiving {} starting with ID {}",
```

Comment:
Nit: This message reads like the swap "starts with" the ID the same way a sentence starts with a specific word.

You might want to start thinking about tracing's [spans](https://docs.rs/tracing/0.1.22/tracing/span/index.html) and attach the ID as a context to all log messages.

Make sure to read the linked docs in detail regarding async code, there are some subtleties to it on how to use it correctly.

Reply:
Do I understand this correctly, that the span would represent the swap (i.e. the period of time one specific swap is executed.)?

Reply:
Yes. You probably want to attach the span using `.instrument` to the `Future` that represents the swap execution and include the ID within that swarm. That will automatically tag all tracing events with this ID if they are logged within that `Future`.

Reply:
I'd keep this in mind until we actually implement concurrent swap.